server/initialize.sql

```diff
@@ -0,0 +1,67 @@
+-- SQL file for making a fresh database with 6 tables
+
+-- drop all tables to completely reset the database
+DROP TABLE if exists regions CASCADE;
+DROP TABLE if exists cohorts CASCADE;
+DROP TABLE if exists users CASCADE;
+DROP TABLE if exists modules CASCADE;
+DROP TABLE if exists weeks CASCADE;
+DROP TABLE IF exists volunteer_flags;
+
+CREATE TABLE regions(
+  id   SERIAL PRIMARY KEY,
+  name VARCHAR(30) NOT NULL
+);
+
+CREATE TABLE cohorts (
+  id                     SERIAL PRIMARY KEY,
+  number                 SMALLINT NOT NULL,
+  region_id              INT NOT NULL,
+  FOREIGN KEY(region_id) REFERENCES regions(id)
+);
+
+CREATE TABLE users (
+  id                     SERIAL PRIMARY KEY,
+  first_name             VARCHAR(20) NOT NULL,
+  last_name              VARCHAR(20) NOT NULL, 
+  pass_hash              TEXT NOT NULL,
+  user_name              VARCHAR(15) NOT NULL,
+  is_volunteer           BOOLEAN NOT NULL,
+  cohort_id              INT,
+  FOREIGN KEY(cohort_id) REFERENCES cohorts(id),
+
+  -- if statement checking if is_volunteer is false then add NOT NULL to cohort_id column 
+  -- this means that if a volunteer signs up (is_volunteer = true) they don't need to provide a cohort_id 
+  -- if a trainee signs up (is_volunteer = false) they will HAVE to provide a cohort_id
+  CONSTRAINT check_volunteer 
+    CHECK ( (is_volunteer) OR (cohort_id IS NOT NULL))
+);             
+
+CREATE TABLE modules (
+  id   SERIAL PRIMARY KEY,
+  name VARCHAR(20) NOT NULL
+);
+
+CREATE TABLE weeks (
+  id                     SERIAL PRIMARY KEY,
+  week_date              DATE NOT NULL,
```

Comment:
Hi Kate, how does the weeks table sit with our data? Does it refer to JS Week-1, JS Week-2, Node Week-1? If it does, the id runs incrementally to N numbers, which can't sit with the above. Or is it a new label we are adding?

Reply:
Yes! So the weeks table would refer to when the weekend classes took place. Has a ``week_date`` for the day it takes place, an ``end_time`` and ``start_time`` so we know the exact time - so we can compare that to the ``volunteer_flags(clockin_time)``. The ``module_id`` is so we can add a module string to that row (JavaScript 1, HTML/CSS, React). 

The `` weeks `` table might have : 

| id | week_date | start_time | end_time | module_id
| --- | --- | --- | --- | --- |
| 1 | 12/03/2022 | 10:00 | 16:00 | 1
| 2 | 19/03/2022 | 10:00 | 16:00 | 2

The `` modules `` table might have : 

| id | name 
| --- | ---
| 1 | 'Final Project Week 2'
| 2 | 'Final Project Week 3'

Hopefully, I understood your question. Please lemme know if you have any more!

Reply:
Oh great! It's clear now. Thanks, Kate.